core/command.go

```diff
@@ -1,13 +1,57 @@
 package core
 
-import "errors"
+import (
+	"errors"
+	"sync"
+)
 
 // State represents system configuration after processing user input
 type State struct {
 	ancestors ancestors
 	Folder    *File
 	Selected  int
 	history   map[*File]int // history of all selected postions
+	*FileMarker
+}
+
+type FileMarker struct {
+	Marked []*File
```

Comment:
I'm not a fan of using maps with arbitrary values, i.e. where we only need to key.
But in this case it will greatly simplify the code. I've decided to use a `map[*File]bool` because we should dynamically generate a file's path. Updated MR.